### PR TITLE
Fixes #6147

### LINKED ISSUE #6147: HTTPS redirection broken if Root CA is a self-signed
If you have certificate installed and issued by the Root CA itself, PF will detect that you have a self signed certificate into the chain and thus disable the HTTPS redirection on the captive portal.

We should exclude the Root CA check within the chain. Not all certificates are issued by Sub CAs.

### PR DESCRIPTION
**Pull request by fdurand**
# Description
Exclude CA cert from isSelfSigned

# Impacts
Secure redirect

# Issue
fixes #6147

# Delete branch after merge
YES

# Checklist
- [ ] Document the feature
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

# NEWS file entries

## Bug Fixes
* HTTPS redirection broken if Root CA is a self-signed (#6147)

### CODE REVIEW

File: lib/pfconfig/namespaces/config/Pf.pm [hunk 1]
```diff
@@ -264,4 +270,4 @@ USA.
 # vim: set shiftwidth=4:
 # vim: set expandtab:
 # vim: set backspace=indent,eol,start:
-
+/
```
Comment (julsemaan) [inline lib/pfconfig/namespaces/config/Pf.pm]: Not sure this slash belongs in here

File: lib/pfconfig/namespaces/config/Pf.pm [hunk 2]
```diff
@@ -221,11 +221,17 @@ sub isSelfSigned {
         $pemcert .= $row;
         if($row =~ /^\-+END(\s\w+)?\sCERTIFICATE\-+$/) {
             my $cert = Crypt::OpenSSL::X509->new_from_string($pemcert);
-            if ($cert->is_selfsigned) {
+            my $exts = $cert->extensions_by_oid();
+            my $ca = $FALSE;
+            foreach my $oid (keys %$exts) {
+                my $ext = $$exts{$oid};
+                $ca = $TRUE if ( ( $oid eq '2.5.29.19' ) && ( $ext->to_string() ~= /CA:TRUE/i ) );
```
Comment (jrouzierinverse) [inline lib/pfconfig/namespaces/config/Pf.pm]: change ```~=``` to ```=~```

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

File: lib/pfconfig/namespaces/config/Pf.pm [hunk 3]
```diff
@@ -221,11 +221,17 @@ sub isSelfSigned {
         $pemcert .= $row;
         if($row =~ /^\-+END(\s\w+)?\sCERTIFICATE\-+$/) {
             my $cert = Crypt::OpenSSL::X509->new_from_string($pemcert);
-            if ($cert->is_selfsigned) {
+            my $exts = $cert->extensions_by_oid();
+            my $ca = $FALSE;
+            foreach my $oid (keys %$exts) {
+                my $ext = $$exts{$oid};
+                $ca = $TRUE if ( ( $oid eq '2.5.29.19' ) && ( $ext->to_string() =~ /CA:TRUE/i ) );
```
Comment (jrouzierinverse) [inline lib/pfconfig/namespaces/config/Pf.pm]: No need to loop just look up the ext by the oid.
```perl
my $ext = $$exts{'2.5.29.19'};
$ca = $TRUE if defined $ext && $ext->to_string() =~ /CA:TRUE/i;
```

**Review by jrouzierinverse — CHANGES_REQUESTED** [review]

**Comment by fdurand** [comment]
BUMP

**Comment by extrafu** [comment]
@jrouzierinverse If all good, merge to get it out of our way for the v11.2 release.

**Comment by extrafu** [comment]
Bump again, until noon or moving out of 11.2

**Pull request merged**